cluster-autoscaler/core/scale_up.go

```diff
@@ -734,8 +741,10 @@ func applyScaleUpResourcesLimits(
 	return newNodes, nil
 }
 
-func scaleUpError(s *status.ScaleUpStatus, err errors.AutoscalerError) (*status.ScaleUpStatus, errors.AutoscalerError) {
+func scaleUpError(s *status.ScaleUpStatus, err errors.AutoscalerError, failedCreationNodeGroup cloudprovider.NodeGroup, failedResizeNodeGroup cloudprovider.NodeGroup) (*status.ScaleUpStatus, errors.AutoscalerError) {
```

Comment:
Do you know why we have this function? It would made sense if it was for adding an error to an existing status object, but it looks like all uses actually create a new status object right in the invocation. Wouldn't it be easier to just directly set all the fields and get rid of the function?

Reply:
Ok, I see. If we want to keep the function, I'd rather keep just the error as an argument (since the repetition is in setting `s.Result = status.ScaleUpError`), and set everything else directly on the status object passed to the function. This way we don't have the `nil, nil` added everywhere where we're not setting these fields, and we don't have to change the function's signature every time we want to set a new field in just one place. WDYT?